### PR TITLE
Create a script in the sandbox testing the boolean data type with Lua

### LINKED ISSUE #1561: Create a script in the sandbox testing the boolean data type with Lua
The custom-scripts team has decided that, along with the documentation we've written up this quarter, we wanted to leave a substantial code base through which future teams can clearly see how they can implement Lua with the different data types described in our arg_custom_type module.

This issue is complete when a successful merge request is made which incorporates a Lua function which can accurately update a boolean data type ingrained in a unique game struct.

### PR DESCRIPTION
**Pull request by tatmaca**
Merge request to resolve Issue #1561:

"The custom-scripts team has decided that, along with the documentation we've written up this quarter, we wanted to leave a substantial code base through which future teams can clearly see how they can implement Lua with the different data types described in our arg_custom_type module.

This issue is complete when a successful merge request is made which incorporates a Lua function which can accurately update a boolean data type ingrained in a unique game struct."

### CODE REVIEW

**Review by RamarkoB — APPROVED** [review]
Looks Great! I already worked with Maggie to finish up the files, and It looks like everything compiles correctly, so the demos and the updated boolean system is working correctly. We can still only compile one file at a time, but that doesn’t seem to be an actual problem! Good work on the new booleans!

File: src/custom-scripts/CMakeCache.txt [hunk 1]
```diff
@@ -0,0 +1,378 @@
+# This is the CMakeCache file.
```
Comment (ecoble) [inline src/custom-scripts/CMakeCache.txt]: Please remove this cache file

File: src/custom-scripts/examples/CMakeCache.txt [hunk 2]
```diff
@@ -0,0 +1,378 @@
+# This is the CMakeCache file.
```
Comment (ecoble) [inline src/custom-scripts/examples/CMakeCache.txt]: Please remove this cache file

File: src/custom-scripts/examples/Makefile [hunk 3]
```diff
@@ -0,0 +1,208 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (ecoble) [inline src/custom-scripts/examples/Makefile]: This should also be removed

File: src/custom-scripts/examples/demonstration.c [hunk 4]
```diff
@@ -69,6 +76,12 @@ int main(int argc, char **argv)
     action_type_t taste_action = {"TASTE", ITEM};
     add_entry(taste_action.c_name, kind1_action_operation, &taste_action, ctx->cli_ctx->table);
 
+    // make key something that can be taken
```
Comment (ecoble) [inline src/custom-scripts/examples/demonstration.c]: Maybe start this comment with a TODO so it is clear you are planning to implement this.

File: src/custom-scripts/examples/flip.lua [hunk 5]
```diff
@@ -0,0 +1,4 @@
+function foo(st)
```
Comment (ecoble) [inline src/custom-scripts/examples/flip.lua]: You can instead just do `return !st`

File: src/custom-scripts/examples/gold.h [hunk 6]
```diff
@@ -0,0 +1,11 @@
+#include <stdlib.h> 
+#include <stdio.h>
+#include "lua/lauxlib.h"
+#include "lua/lua.h"
+#include "lua/lualib.h"
+#include "lua/lapi.h"
+
+typedef struct chest
+{
+int coins;
```
Comment (ecoble) [inline src/custom-scripts/examples/gold.h]: Indentation

**Review by ecoble — CHANGES_REQUESTED** [review]
Looks great overall! A few small changes, also it looks like there are still some cache files. Please refer to the message Borja sent about removing them, if you are having trouble with it let me know.

**Review by ecoble — CHANGES_REQUESTED** [review]
Also please remove the Lua submodule, it should get included by the build

**Review by ecoble — CHANGES_REQUESTED** [review]
Please remove the Lua submodule, it should already be installed on the CSIL machines. Let me know if you are having trouble doing this.

**Review by ecoble — APPROVED** [review]

**Comment by tatmaca** [comment]
This issue is now closed! In preparation for our final demo, we've merged a branch that runs a chiventure game and experiments with using Lua to manipulate boolean values. This simple function is critical as it can add easy game manipulation through the use of ON/OFF, ACTIVE/INACTIVE, etc. to implement logic on the players end as they progress through a game. This demo can be found in dark_room.c.

**Comment by ecoble** [comment]
Issue Score: Excellent

Comments:
Great job on this example, and nice work making the requested changes!

**Pull request merged**